gridstatus/ieso.py

```diff
@@ -0,0 +1,473 @@
+        # Multiply the interval minus 1 by 5 to get the minutes in the range 0-55
+        df["Interval Minute Start"] = MINUTES_INTERVAL * (df["Interval"] - 1)
+        df["Published Time"] = pd.Timestamp(created_at, tz=self.default_timezone)
+
+        df["Interval Start"] = (
+            df["Delivery Date"]
+            + pd.to_timedelta(df["Delivery Hour Start"], unit="h")
+            + pd.to_timedelta(df["Interval Minute Start"], unit="m")
+        )
+
+        df["Interval End"] = df["Interval Start"] + pd.Timedelta(
+            minutes=MINUTES_INTERVAL,
+        )
+
+        cols_to_keep_in_order = [
+            "Interval Start",
+            "Interval End",
+            "Published Time",
+            "Market Total Load",
+            "Ontario Load",
+        ]
+
+        df = utils.move_cols_to_front(df, cols_to_keep_in_order)[cols_to_keep_in_order]
+
+        if end:
+            return df[df["Interval End"] <= pd.Timestamp(end)]
+
+        return df
+
+    def get_load(self, date, end=None, verbose=False):
+        """
+        Get hourly load for the Market and Ontario for a given date or from
+        date to end date.
+
+        Args:
+            date (datetime.date | datetime.datetime | str): The date to get the load for
+                Can be a `datetime.date` or `datetime.datetime` object, or a string
+                with the values "today" or "latest". If `end` is None, returns
+                only data for this date.
+            end (datetime.date | datetime.datetime, optional): End date. Defaults None
+                If provided, returns data from `date` to `end` date. The `end` can be a
+                `datetime.date` or `datetime.datetime` object.
+            verbose (bool, optional): Print verbose output. Defaults to False.
+
+
+        Returns:
+            pd.DataFrame: zonal load as a wide table with columns for each zone
+        """
+        if isinstance(date, tuple):
+            date, end = date
+        data_five_minutes = self.get_5_min_load(date, end, verbose)
+
+        # Hourly demand is the average over the 5 minute intervals within each hour
+        df = data_five_minutes.groupby(
+            [
+                data_five_minutes["Interval Start"].dt.date,
+                data_five_minutes["Interval Start"].dt.hour,
+            ],
+        ).agg(
+            {
+                "Market Total Load": "mean",
+                "Ontario Load": "mean",
+                "Interval Start": "min",
+            },
+        )
+
+        df["Interval End"] = df["Interval Start"] + pd.Timedelta(hours=HOUR_INTERVAL)
+
+        df = utils.move_cols_to_front(
+            df,
+            [
+                "Interval Start",
+                "Interval End",
+                "Market Total Load",
+                "Ontario Load",
+            ],
+        )
+
+        return df.reset_index(drop=True)
+
+    @support_date_range(frequency="DAY_START")
+    def get_load_forecast(self, date, end=None, verbose=False):
```

Comment:
i think we need to have 2 different load forecasts methods 

* the one you implement already using [this](http://reports.ieso.ca/public/OntarioZonalDemand/PUB_OntarioZonalDemand.xml) should be `get_ zonal_load_forecast`

* get_load_forecast(date="latest") should return the latest load forecast available using https://www.ieso.ca/-/media/Files/IESO/Power-Data/Ontario-Demand-multiday.ashx. i think this is better the xml file linked by the graph since it doesnt included the 5 minute data, which we dont need here

Reply:
For `get_load_forecast`, that means we'll only have the forecast for Ontario and will not be offering the Market load forecast?

Reply:
correct. i dont see an better alternative right now. we can revisit in future

Reply:
Got it. And `get_load_forecast` only supports forecasts for today? To get a forecast for future dates, users will need to call `get_zonal_load_forecast`.

Reply:
Sorry, I just saw that [link](https://www.ieso.ca/-/media/Files/IESO/Power-Data/Ontario-Demand-multiday.ashx) has data through tomorrow.